cmd/merge/clickhouse.go

```diff
@@ -0,0 +1,97 @@
+package cmd_merge //nolint:stylecheck
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+	"github.com/flashbots/mempool-dumpster/common"
+	"go.uber.org/zap"
+)
+
+var ErrNoDSN = fmt.Errorf("Clickhouse DSN is required")
+
+type ClickhouseOpts struct {
+	Log *zap.SugaredLogger
+	DSN string
+}
+
+type Clickhouse struct {
+	opts ClickhouseOpts
+
+	log  *zap.SugaredLogger
+	conn driver.Conn
+}
+
+// NewClickhouse creates a new Clickhouse instance with a connection to the database.
+func NewClickhouse(opts ClickhouseOpts) (*Clickhouse, error) {
+	ch := &Clickhouse{
+		log:  opts.Log,
+		opts: opts,
+	}
+	if ch.opts.DSN == "" {
+		return nil, ErrNoDSN
+	}
+
+	err := ch.connect()
+	if err != nil {
```

Comment:
stylistic optional nit:


```suggestion
	if err := ch.connect(); err != nil {
```

Reply:
fine with me